### PR TITLE
AppConfig entity and endpoints

### LINKED ISSUE #413: Create AppConfig entity and API endpoints
### Pre-requisites:
- [x] I looked through both [open and closed issues](../issues?utf8=✓&q=is%3Aissue) and did not find another request for the same or similar feature.

## Description

We need a new table which contains app configuration information that can be retrieved by other parts of the app.

`POST`, `PUT`, and `GET` endpoints should be created and should only be accessible by the admins (dependent on #398)

### PR DESCRIPTION
**Pull request by Ulfingrad**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

## Description of Changes

Adds the `AppConfig` entity in addition to adding endpoints to interact with said entity.

## Related Issues

Closes #413

### CODE REVIEW

File: packages/api/src/api/appConfig.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { AppConfig } from '../entities/AppConfig';
+import { populateUser } from '../middleware/populateUser';
+
+export const appConfig = Router();
+appConfig.use(populateUser());
+
+appConfig.get('/', async (req, res) => {
+  const user = req.userEntity;
+  if (!user.isAdmin) {
+    res.status(403).send('only admins have permission to access this endpoint!');
```
Comment (SpencerKaiser) [inline packages/api/src/api/appConfig.ts:12]:
```suggestion
    res.sendStatus(403);
```

File: packages/api/src/api/appConfig.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    res.status(500).send(errorMsg);
+  }
+});
+
+appConfig.post('/', async (req, res) => {
+  const user = req.userEntity;
+  const { key, value } = req.body;
+
+  if (!user.isAdmin) {
+    res.status(403).send('only admins have permission to access this endpoint!');
```
Comment (SpencerKaiser) [inline packages/api/src/api/appConfig.ts:59]: Same thing for the rest of these

File: packages/api/src/entities/AppConfig.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+export enum ConfigKey {}
+
+type JSONValue = string | number | boolean | Record<string, any> | any[];
+
+export type AppConfigConstructorValues = ConstructorValues<AppConfig>;
+type AppConfigPropertyKeys = keyof AppConfigConstructorValues;
+
+@Entity()
+export class AppConfig extends Node<AppConfig> {
+  @Property({ columnType: 'text' })
```
Comment (SpencerKaiser) [inline packages/api/src/entities/AppConfig.ts:15]: Make this unique

File: packages/api/src/api/appConfig.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  try {
+    const configItemToUpdate = await req.entityManager.findOne(AppConfig, { key });
+
+    if (!configItemToUpdate) {
+      res.status(400).send('The specified config item does not exist');
+      return;
+    }
+
+    configItemToUpdate.value = value;
+    await req.entityManager.persistAndFlush(configItemToUpdate);
```
Comment (SpencerKaiser) [inline packages/api/src/api/appConfig.ts:44]: After making `key` unique, simply create a new `AppConfig` and try `persistAndFlush`

File: packages/api/src/entities/AppConfig.ts [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+export enum ConfigKey {}
+
+type JSONValue = string | number | boolean | Record<string, any> | any[];
+
+export type AppConfigConstructorValues = ConstructorValues<AppConfig>;
+type AppConfigPropertyKeys = keyof AppConfigConstructorValues;
+
+@Entity()
+export class AppConfig extends Node<AppConfig> {
```
Comment (SpencerKaiser) [inline packages/api/src/entities/AppConfig.ts:14]: @johnkahn if we don't extend `Node` here and simply create a brand new entity, will `persistAndFlush` act as an `upsert` when we try and save a new `AppConfig` with the same `key`?

**Review by dre8597 — APPROVED** [review]
LGTM 🚀

**Pull request merged**